client/src/main/resources/process-0.1.xsd

```diff
@@ -331,6 +338,40 @@
             <xs:enumeration value="oozie"/>
             <xs:enumeration value="pig"/>
             <xs:enumeration value="hive"/>
+            <xs:enumeration value="spark"/>
+        </xs:restriction>
+    </xs:simpleType>
+
+    <xs:complexType name="spark-attributes">
+        <xs:annotation>
+            <xs:documentation>
+                List of Spark attributes to run Spark application:
+                    master element indicates the url of the Spark Master. Ex: client, yarn-master, or local.
+                    mode element if present indicates the mode of spark, where to run spark driver program. Ex: client,cluster.
+                    name element indicates the name of the spark application.
+                    application-type indicates type of the spark application java, python.
+                    class element if present, indicates the spark's application main class.
+                    jar element indicates a comma separated list of jars or python files.
+                    spark-opts element if present, contains a list of spark options that can be passed to spark driver.
+                    arg element if present, contains arguments that can be passed to spark application.
+            </xs:documentation>
+        </xs:annotation>
+        <xs:sequence>
+            <xs:element name="master" type="xs:string" minOccurs="1" maxOccurs="1"/>
```

Comment:
I think it may be good to have master in the cluster spark interface and overridable using the process properties as needed